examples/technical_blogs/ock_demo_blog/getting_started.md

```diff
@@ -0,0 +1,528 @@
+
+We will focus on ```clVectorAddition``` in this subsection.
+
+Here's how to run ```clVectorAddition```:
+
+```sh
+    cd $RELEASE_DIR/build_riscv_tests/bin
+    CA_HAL_DEBUG=1 CA_PROFILE_LEVEL=3 OCL_ICD_FILENAMES=$RELEASE_DIR/install/lib/libCL.so ONEAPI_DEVICE_SELECTOR=opencl:acc SYCL_CONFIG_FILE_NAME=  ./clVectorAddition
+```
+
+This is the expected output you should have:
+
+```
+1: Test timeout computed to be: 10000000
+1: refsi_hal_device::mem_alloc(size=256, align=128) -> 0xb800fe80
+1: refsi_hal_device::mem_alloc(size=256, align=128) -> 0xb800fd80
+1: refsi_hal_device::mem_alloc(size=256, align=128) -> 0xb800fc80
+1: refsi_hal_device::mem_write(dst=0xb800fe80, size=256)
+1: refsi_hal_device::mem_write(dst=0xb800fd80, size=256)
+1: refsi_hal_device::program_find_kernel(name='vector_addition.mux-kernel-wrapper') -> 0x00010000
+1: refsi_hal_device::kernel_exec(kernel=0x00010000, num_args=3, global=<64:1:1>, local=<64:1:1>)
+1: refsi_hal_device::mem_read(src=0xb800fc80, size=256)
+1: refsi_hal_device::mem_free(address=0xb800fe80)
+1: refsi_hal_device::mem_free(address=0xb800fd80)
+1: refsi_hal_device::mem_free(address=0xb800fc80)
+1: Available platforms are:
+1:   1. ComputeAorta
+1: 
+1: Selected platform 1
+1: 
+1: Running example on platform 1
+1: Available devices are:
+1:   1. RefSi M1
+1: 
+1: Selected device 1
+1: 
+1: Running example on device 1
+1:  * Created context
+1:  * Built program
+1:  * Created buffers
+1:  * Created kernel and set arguments
+1:  * Created command queue
+1:  * Enqueued writes to source buffers
+1:  * Enqueued NDRange kernel
+1:  * Enqueued read from destination buffer
+1:  * Result verified
+1:  * Released all created OpenCL objects
+1: 
+1: Example ran successfully, exiting
+1: [+] total retired instructions: 98
+1: [+] total elapsed cycles: 98
+1: [+] total direct memory write access: 1KB
+1: [+] total direct memory read access: 256B
+```
+
+## Debugging, profiling and examining low level instructions
+
+The following subsections describe how to debug, profile and examine the low
+level instructions emitted during program execution using the samples and
+benchmarks we previously built.
+
+The following examples will be using the ```simple-vector-add``` kernel from
+the RISCV-examples directory.
+
+```sh
+    cd $RELEASE_DIR/RISCV-examples/bin
+```
+
+### Profiler
+
+First, check you are in the right directory to run the sample:
+
+```sh
+    cd $RELEASE_DIR/RISCV-examples/bin
+```
+
+The profiler is a built-in feature that enables you to profile or track
+performance of a kernel. It will display statistics gathered during the
+execution of that kernel.
+
+There are several ways that you can enable the profiler and store those
+information.
+
+Here's the variables that will be used:
+- ```CA_PROFILE_LEVEL```
+- ```CA_PROFILE_CSV_PATH```
+
+#### CA_PROFILE_LEVEL
+
+The profiler has four distinct levels to perform tracking. Here's how to set
```

Comment:
we've kind of mentioned CA_PROFILE_LEVEL above, should we drop it from there or reorder - maybe do the networks last?